relayer/src/channel.rs

```diff
@@ -1191,3 +1268,51 @@ fn check_destination_channel_state(
         )))
     }
 }
+
+/// Queries a channel end on a [`ChainHandle`], and verifies
+/// that the counterparty of that channel matches an
+/// expected counterparty.
+/// Returns `Ok` if the counterparty matches, and `Err` otherwise
+pub fn check_channel_counterparty(
```

Comment:
Do we want to move this to the [`chain::counterparty`](https://github.com/informalsystems/ibc-rs/blob/master/relayer/src/chain/counterparty.rs) perhaps?

Reply:
Good idea! I moved the method in that module.

7243c93a4855fc8e8724d9ba3a8dd67b4657fdd5

As a side note, most methods inside `chain::counterparty` return in case of error the type [`supervisor::Error`](https://github.com/informalsystems/ibc-rs/blob/26ef2c6fcc2527998be1a206292ad870191b42c0/relayer/src/supervisor/error.rs#L7), whereas the method I just added uses the type [`ChannelError`](https://github.com/informalsystems/ibc-rs/blob/26ef2c6fcc2527998be1a206292ad870191b42c0/relayer/src/channel.rs#L35). I think the other methods in this module should also use an error type that is more specific (nothing to do with the supervisor), but will not address this issue in this PR (will keep it in mind for a minor PR later).

Reply:
Good catch! Agreed